core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/migration/WorkspaceMigrationActor.scala

```diff
@@ -439,31 +431,33 @@ object WorkspaceMigrationActor {
         )
 
         _ <- MigrateAction.fromFuture {
+          import SamBillingProjectPolicyNames.owner
           for {
-            accessPolicies <- samDao
-              .admin
-              .listPolicies(
-                SamResourceTypeNames.workspace,
-                workspace.workspaceId,
-                userInfo
-              )
-              .map(_.map(p => p.policyName -> p.email).toMap)
-
-            // billing project owners get special billing project owner-y iam roles on the
-            // workspace's google project. See the link below for more info.
-            // https://docs.google.com/document/d/1uOGtmw_l_Ve2gqJohLGntJmQz69md2h3T2EAvyKTm20
-            billingProjectOwnerPolicyEmail = accessPolicies
-              .getOrElse(projectOwner, throw WorkspaceMigrationException(
-                message = s"""Workspace migration failed: no "$projectOwner" policy on workspace.""",
-                data = Map("migrationId" -> migration.id, "workspace" -> workspace.toWorkspaceName)
+            billingProjectPolicies <- samDao.admin.listPolicies(
+              SamResourceTypeNames.billingProject,
+              workspace.namespace,
+              userInfo
+            )
+
+            billingProjectOwnerPolicy = billingProjectPolicies
+              .find(_.policyName == owner)
+              .getOrElse(throw WorkspaceMigrationException(
+                message = s"""Workspace migration failed: no "$owner" policy on billing project.""",
+                data = Map("migrationId" -> migration.id, "billingProject" -> workspace.namespace)
               ))
 
+            workspacePolicies <- samDao.admin.listPolicies(
+              SamResourceTypeNames.workspace,
+              workspace.workspaceId,
+              userInfo
+            )
+
             _ <- workspaceService.setupGoogleProjectIam(
               googleProjectId,
-              accessPolicies,
-              billingProjectOwnerPolicyEmail
+              workspacePolicies.map(p => p.policyName -> p.email).toMap,
+              billingProjectOwnerPolicy.email
```

Comment:
This comes from https://github.com/broadinstitute/rawls/blob/39cbdba87097d0be3b798c2dcb2f1f2ba4b748e8/core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala#L2445, and that email is used by `setupGoogleProjectIam` in the same function, before the code you linked. Let me know what to do because I'm finding it really hard to unpack that function.

Reply:
Done when updating bucket Iam